### PR TITLE
Run Verify Issue linking workflow only when label is bug or enhancement

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Run Verify Issue linking workflow only when PR label is bug or enhancement

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by rohitgr7** [comment]
> Or just do what I always do. Add this to your PR template:
> 
> image
> 
> sweat_smile

hackerman

File: .github/workflows/ci_verify_pr_contains_issue.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
   verify_linked_issue:
     runs-on: ubuntu-latest
     name: Ensure Pull Request has a linked issue.
-    if: github.event.pull_request.draft == false
+    if: github.event.pull_request.draft == false and github.event.label.name == 'enhancement'
```
Comment (rohitgr7) [inline .github/workflows/ci_verify_pr_contains_issue.yaml:13]: enhancement will cover up `bugfix`?

Comment (kaushikb11) [reply .github/workflows/ci_verify_pr_contains_issue.yaml:13]: Naa, It's still Draft and testing it out.

Comment (rohitgr7) [reply .github/workflows/ci_verify_pr_contains_issue.yaml:13]: oh okay! I saw everyone approved so thought maybe it's ready 😄

**Comment by tchaton** [comment]
Hey @kaushikb11.

The simplest would be to fork the initial repo and add our own logic there. We could add this only for us and F.

Best,
T.C

**Comment by Borda** [comment]
@kaushikb11 still a draft? :]

**Pull request closed without merging**